src/plugins/vis_type_timeseries/server/routes/post_vis_schema.ts

```diff
@@ -253,22 +252,19 @@ export const visPayloadSchema = Joi.object({
     })
   ),
   // general
-  query: Joi.array()
-    .items(queryObject)
-    .allow(null)
-    .required(),
-  state: Joi.object({
-    sort: Joi.object({
-      column: stringRequired,
-      order: Joi.string()
-        .valid(['asc', 'desc'])
-        .required(),
-    }).optional(),
-  }).required(),
-  savedObjectId: Joi.string().optional(),
-  timerange: Joi.object({
+  query: schema.maybe(schema.arrayOf(schema.nullable(queryObject))),
```

Comment:
Looks like the query is required, so the `maybe` should be removed here. Also when I'm reading the Joi schema correctly, `query` is allowed to be null, not the items within the array. That's not the same thing, Joi would allow `query: null` but fail on `query: [null]` while the new version would do the exact opposite.

So it should be `schema.nullable(schema.arrayOf(queryObject))`